### DISCUSSION TITLE
singularity containment on hpc cluster & library loading issues

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#329] [Q&A]
We try to run DIA-NN on a hpc computing cluster. This requires a singularity containment solution. 
We manage to execute DIA-NN via the command line tool and load raw files and fastas but not libraries. 
So far, we tried .xls libraries exported from Biognosys and .tsv libraries and .tsv libraries exported from the DIA-NN GUI itself. 
We get the following error message:

 
![Capture](https://user-images.githubusercontent.com/98888664/156742833-3aff436d-41c9-420d-a004-bd6365dbb958.JPG)

**Comment by vdemichev** [comment]
Looks like permissions problem on the specific Linux machine. DIA-NN just uses the regular C++ interface for opening files, if it fails it means the respective system command failed, most likely because of permissions.

Vadim

**Reply by vdemichev** [reply]
This is the code that reads the .tsv/.csv/.xls libraries: 
![image](https://user-images.githubusercontent.com/30595084/157746208-e15edd7b-9ad2-4d90-9698-d64ccc277819.png)

**Reply by vdemichev** [reply]
And this is for reading a FASTA database:
![image](https://user-images.githubusercontent.com/30595084/157748721-9bd76d6f-f136-410d-bcf9-4ca669f743dc.png)

**Reply by vdemichev** [reply]
The code is basically the same, ifstream salways has implicit 'in' flag set, so std::istream::in here has no effect. Don't know why one faisl and the other one works...

**Comment by JM-Bader** [comment]
The files have permissions as shown in the screenshot, i.e. it should not be a problem. 

We thus tried it on another linux computer (not a singularity containment) with the identical files copied there. There it worked. 

Thus we explored options within the singularity containment on the cluster again. Initially (screenshot above), all files were on a network drive. DIA-NN could be executed from there and managed to load the raw file and the fasta file - however produced the library error message. 

When executing DIA-NN from the home directory and having all files there (raw file, fasta, library), it worked. 

Surprisingly, it also ran through when DIA-NN was executed from the network drive, the raw file and the fasta file were loaded from the network drive but the library was loaded from the home directory.
> Why is that the case?

In any case, my home directory is too small to accomodate all raw files I try to analyze - so they need to be on a network drive. [Our IT/server infrastructure is designed in a way that users are supposed to use network drives for large datasets]. Is there a disadvantage (other than potentially slower loading via the network) if the raw files are on a network drive?

**Reply by vdemichev** [reply]
Fine with network drives except (i) speed (ii) not sure how signficant network delays, which might occur, will be handled by the C++ input/output, whether delays will produce an error.
